tools/clang-tidy-plugin/CMakeLists.txt

```diff
@@ -58,7 +58,7 @@ else ()
 endif ()
 
 target_include_directories(${CataAnalyzerName} SYSTEM PRIVATE
-        ${LLVM_INCLUDE_DIRS} ${CLANG_INCLUDE_DIRS})
+        ${LLVM_INCLUDE_DIRS} ${CLANG_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR}/../../src/third-party)
```

Comment:
This line is for header search path for the custom clang-tidy plugin. No need to add ImGui there.

Reply:
Clang-tidy check is currently failing because point.h header includes ImGui.h, I was trying to fix that. How else can I fix that?

Reply:
The CMake script for the main game (I guess it's `src/CMakeLists.txt`) needs to be fixed, not the CMake script for clang-tidy plugin.

Reply:
Remove the imgui include from point.h and [these two functions](https://github.com/CleverRaven/Cataclysm-DDA/blob/f9eafbe099ba85d50339de2dd78ed21e8b71d324/src/point.h#L99-L102). They're only used for an implicit conversion [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/f9eafbe099ba85d50339de2dd78ed21e8b71d324/src/cata_imgui.cpp#L306) that clang-tidy won't like anyway ([`google-explicit-constructor`](https://clang.llvm.org/extra/clang-tidy/checks/google/explicit-constructor.html)). Replace it with an explicit function, local to cata_imgui.cpp, that converts the point type instead.

Reply:
OK. deleted that constructor and added conversion functions in cata_imgui.h that don't require me to include imgui.h from that file either.

Reply:
I'll probably rebase tomorrow to remove the commit to the clang-tidy CMakeFiles.txt

Reply:
Nevermind, decided to force-push just now. This is fixed.